### DISCUSSION TITLE
Is there a way to save outside of cache?

### DISCUSSION THREAD

**Discussion by tabularelf** [rocksdanister/lively#1658] [Q&A]
The title itself doesn't really explain enough, but I'm basically working on a new interactive wallpaper that saves states. You have a plant that you maintain and grow, but in order to have it grow in stages, even between Lively restarts, I need to be able to save it.

Though I've noticed whether I use Web or an Application, the save file that I've made doesn't actually get saved. It's like ending up in a cache sandbox, and gets removed once Lively closes/the wallpaper changes. Running it by itself doesn't have this issue, and neither on Wallpaper Engine. Is this a bug, or by design?

(I was gonna make this as an issue, but I wanted to see if it was a bug or not first, or if there's anything I could do to work around this.)

**Comment by rocksdanister** [comment]
<img width="700" alt="image" src="https://user-images.githubusercontent.com/17554161/229969171-9300c4b4-5cb7-4205-afd7-232a579171a8.png">

Default is memory cache.

**Reply by rocksdanister** [reply]
Looks like I am only allowing it for url and not html file:
https://github.com/rocksdanister/lively/blob/c27d2d04e9d4e921c83ba74465e0869402e4fc83/src/Lively/Lively/Core/Wallpapers/WebCefSharpProcess.cs#L67

**Comment by tabularelf** [comment]
Ahhh yeah, that's a lil problematic haha. Could you add in support for local HTML too?
I did also do disk caching in the settings prior to this, but that didn't work for me either, I forgot to mention that! Doing it with URL works, but not ideal for me.
